src/integrationtests/java/com/aws/greengrass/integrationtests/deviceauth/PolicyWildcardTest.java

```diff
@@ -0,0 +1,243 @@
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.aws.greengrass.AuthorizeClientDeviceActionResponseHandler;
+import software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCClient;
+import software.amazon.awssdk.aws.greengrass.model.AuthorizeClientDeviceActionRequest;
+import software.amazon.awssdk.aws.greengrass.model.AuthorizeClientDeviceActionResponse;
+import software.amazon.awssdk.eventstreamrpc.EventStreamRPCConnection;
+
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.aws.greengrass.testcommons.testutilities.ExceptionLogProtector.ignoreExceptionOfType;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@SuppressWarnings("PMD.AvoidCatchingGenericException")
+@ExtendWith({GGExtension.class, UniqueRootPathExtension.class, MockitoExtension.class})
+class PolicyWildcardTest {
+
+    @TempDir
+    Path rootDir;
+    Kernel kernel;
+
+    Certificate certificate;
+
+    String clientPem;
+
+    @BeforeEach
+    void beforeEach(ExtensionContext context) throws Exception {
+        ignoreExceptionOfType(context, SpoolerStoreException.class);
+        ignoreExceptionOfType(context, NoSuchFileException.class); // Loading CA keystore
+
+        // Set this property for kernel to scan its own classpath to find plugins
+        System.setProperty("aws.greengrass.scanSelfClasspath", "true");
+        kernel = new Kernel();
+
+        // Set up Iot auth client
+        IotAuthClientFake iotAuthClientFake = new IotAuthClientFake();
+        kernel.getContext().put(IotAuthClient.class, iotAuthClientFake);
+
+        // start CDA service with configuration
+        startNucleusWithConfig("wildcard-config.yaml");
+
+        // create certificate that client devices can use
+        setClientDeviceCertificatePem();
+    }
+
+    @AfterEach
+    void afterEach() {
+        LogConfig.getRootLogConfig().reset();
+        kernel.shutdown();
+    }
+
+    @Test
+    void GIVEN_policy_with_inline_wildcards_WHEN_publish_authz_THEN_authorized() throws Exception {
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel, "main")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+
+            String deviceToken = getClientDeviceSessionAuthToken("myThing", clientPem);
+
+            Map<AuthorizeClientDeviceActionRequest, Boolean> testCases = new HashMap<>();
+            // valid requests
+            testCases.put(
```

Comment:
nit: this is my bad, can we make this a @ParameterizedTest, idk why i did it in the test itself

Reply:
I'll update this

Reply:
done